### ISSUE TITLE
Certain GCP resources do not work due to naming difference between regions

### ISSUE THREAD

**Issue by jakubserwatka-oktawave** [infracost/infracost#2752]
For example:

**Static IPs**

in internal/resources/google/compute_address.go
```
AttributeFilters: []*schema.AttributeFilter{
  {Key: "description", Value: strPtr("Static Ip Charge")},
},
```

However, in europe-central2 this resource is called `Static Ip Charge in Warsaw` and has different SKU

So Infracost will report $0.00 as cost

This situation is true for multiple regional resources in GCP especially the newer regions

I rebuild the binary with 

```
AttributeFilters: []*schema.AttributeFilter{
  {Key: "description", Value: strPtr("Static Ip Charge in Warsaw")},
},
```

and the cost for my IP address provisioned in europe-central2 was reported correctly

**Comment by vdmgolub** [comment]
@jakubserwatka-oktawave Hey! Thanks a lot for creating the issue! I've confirmed that's an issue for the `google_compute_address` resource. Tweaking the attribute filter should accommodate all the cases, I've created a PR for that.

You mentioned that other resources can have a similar issue too. Could you please list them here if you encountered them here? We're not using Google resources ourselves, especially across different regions, so we rely on the community to report such cases. I really appreciate any help you can provide. :)

**Comment by jakubserwatka-oktawave** [comment]
I'll go over all existing resources for GCP and verify that there are no other existing cases like this. From quick check I can see that there are some resources that handle it correctly, for example `bigquery_dataset.go` with

```
AttributeFilters: []*schema.AttributeFilter{
  {Key: "description", Value: strPtr(fmt.Sprintf("Analysis (%s)", r.Region))},
},
```

I will also check if this separation of pricing is consistent for any specific region, that way it can be added to tests and suggested for any new resources.

**Comment by JakubSerwatka** [comment]
Hey, didn't have much time too look at it so I skipped more complicated resources (Compute Engine + Networking) and focused on the simpler ones. Here are my finds so far.

**Artifact Registry**
- Africa continent missing in network traffic calculation

**BigQuery**
- Missing support for `location` argument

**BigQuery Dataset**
- Multi-region not supported

**BigQuery Table**
- Multi-region not supported for multiple cost components
- Potential incorrect region attribution due to `r.mapRegion()`
- Storage read API appears to not support regions correctly (judgement based on non regexed attribute filter)

**Cloudfunctions**
- Missing 2nd gen support

**Compute Image**
- Missing support for `storage_locations` argument 

**Compute Snapshot**
- Missing support for `storage_locations` argument

As far as the testing region goes it seams that usually the newer ones get special treatment but that is not a guarantee. I'd suggest testing at least one region from every continent, as well as multi regional resources. Aditionally I suggest testing at least one non-provider default region, they arent always attributed correctly due to non `region` arguments in resources.

**Comment by vdmgolub** [comment]
Thank you for digging into it, great stuff! ❤️  I'll bring your suggestions to the team to improve the test coverage for different regions.

The fix for the compute address is in the master already. We'll keep this issue open until the new release is out. We'll post a comment, so you should be notified when you can upgrade. Thanks again for all your help! :)

**Comment by aliscott** [comment]
@jakubserwatka-oktawave thanks for reporting this 🙏 The fix for compute address is now released in [v0.10.31](https://github.com/infracost/infracost/releases/tag/v0.10.31).
